front_end/models/react_native/FuseboxBindingsModel.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import * as SDK from '../../core/sdk/sdk.js';
+import type * as Common from '../../core/common/common.js';
+
+type JSONValue = null | string | number | boolean | {[key: string]: JSONValue} | JSONValue[];
+type DomainName = 'react-devtools';
+type DomainMessageListener = (message: JSONValue) => void;
+type BindingCalledEventTargetEvent = Common.EventTarget.EventTargetEvent<SDK.RuntimeModel.EventTypes[SDK.RuntimeModel.Events.BindingCalled]>;
+
+const RUNTIME_GLOBAL = '__FUSEBOX_REACT_DEVTOOLS_DISPATCHER__';
+
+export class FuseboxBindingsModel extends SDK.SDKModel.SDKModel {
+  private readonly domainToListeners: Map<DomainName, Set<DomainMessageListener>> = new Map();
+
+  constructor(target: SDK.Target.Target) {
+    super(target);
+
+    this.initialize(target);
+  }
+
+  private initialize(target: SDK.Target.Target): void {
+    const runtimeModel = target.model(SDK.RuntimeModel.RuntimeModel);
+    if (!runtimeModel) {
+      return;
+    }
```

Comment:
When does `!runtimeModel` happen? Will this`FuseboxBindingsModel` be called again in the future when a`RuntimeModel` is ready?

Reply:
I don't really know the whole lifecycle here, but I think this is some rare case and all of these checks are just for type-safety

Looking at other models implementation, they follow the same pattern